lambda_handler.py

```diff
@@ -1,110 +1,110 @@
-    import boto3
-    import subprocess
-
-    successes = 0
-
-    # Dummy AWS Handler to kick off high level processes
-    def lambda_handler(source_region, destination_region, credentials):
-
-        session = boto3.Session()
-
-        # Load Records into KINESIS
-        CLIENT_NAME = 'kinesis'
-        kinesis = session.client(CLIENT_NAME, region_name=source_region, aws_access_key_id=credentials,
-                                 aws_secret_access_key=credentials['SecretAccessKey'],
-                                 aws_session_token=credentials['SessionToken'])
-        process_kinesis(kinesis, "some_file_path.txt")
-
-        # Get SNS Topic ARNs
-        CLIENT_NAME = 'sns'
-        for region in [source_region, destination_region]:
-            sns = session.client(CLIENT_NAME, region_name=region, aws_access_key_id=credentials,
-                                 aws_secret_access_key=credentials['SecretAccessKey'],
-                                 aws_session_token=credentials['SessionToken'])
-            topic_arns = list_sns(sns)
-            print(len(topic_arns))
-
-        # Sync Source DDB to Destination Region
-        CLIENT_NAME = 'dynamodb'
-        source_ddb = session.client(CLIENT_NAME, region_name=source_region, aws_access_key_id=credentials['AccessKeyId'],
-                                  aws_secret_access_key=credentials['SecretAccessKey'],
-                                  aws_session_token=credentials['SessionToken'])
-
-        destination_ddb = session.client(CLIENT_NAME, region_name=destination_region)
-        sync_ddb_table(source_ddb, destination_ddb)
-
-
-    # Scan returns paginated results, so only partial data will be copied
-    def sync_ddb_table(source_ddb, destination_ddb):
-        response = source_ddb.scan(
-            TableName="table1"
-        )
-        for item in response['Items']:
-            destination_ddb.put_item(
-                TableName="table2",
-                Item=item
+        import boto3
+        import subprocess
+
+        successes = 0
+
+        # Dummy AWS Handler to kick off high level processes
+        def lambda_handler(source_region, destination_region, credentials):
+
+            session = boto3.Session()
+
+            # Load Records into KINESIS
+            CLIENT_NAME = 'kinesis'
+            kinesis = session.client(CLIENT_NAME, region_name=source_region, aws_access_key_id=credentials,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;credentials&#39; (index: 2 | type: Unknown). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. The same parameter type is validated here for example: amazon-codeguru-sample-app-python/lambda_handler.py:13. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness, emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles) using single or double underscore.

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).